addons/website/static/src/js/content/website_root.js

```diff
@@ -82,6 +91,35 @@ var WebsiteRoot = publicRootData.PublicRoot.extend({
         }
         return registry;
     },
+    /**
+     * Toggles the fullscreen mode.
+     *
+     * @private
+     * @param {Boolean} state toggle fullscreen on/off (true/false)
+     */
+    async _toggleFullscreen(state) {
+        this.isFullscreen = state;
+        document.body.classList.toggle('o_fullscreen', this.isFullscreen);
+        document.body.style.overflowX = 'hidden';
+        let resizing = true;
+        window.requestAnimationFrame(function resizeFunction() {
+            window.dispatchEvent(new Event('resize'));
+            if (resizing) {
+                window.requestAnimationFrame(resizeFunction);
+            }
+        });
+        document.body.addEventListener('transitionend', function onTransitionEnd(ev) {
+            if (ev.target === document.body && ev.propertyName === 'padding-top') {
+                resizing = false;
+                document.body.style.overflowX = '';
+                document.body.removeEventListener('transitionend', onTransitionEnd);
+            }
+        });
+        // Safeguard in case the transitionend event doesn't trigger for whatever reason.
+        window.setTimeout(() => {
+            resizing = false;
+        }, 500);
```

Comment:
In that potential case, the line `document.body.style.overflowX = '';` should still be called. Maybe it should be put in a `else` clause in the `requestAnimationFrame` loop?

Reply:
It should probably also remove the eventListener for transitionend, now that I think about it. I can just call the transitionend handler directly with a fake event to make sure that the same things happen in both cases.

Reply:
No fake event, just a common function to be called.